src/Cake.Core/Utilities/FastHash.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Cake.Core.Utilities
+{
+    /// <summary>
+    /// Optimized hash generator. Using SHA1CryptoServiceProvider since it is FIPS compliant.
+    /// </summary>
+    public static class FastHash
```

Comment:
Hashing is only needed for Caching at this point, move to Cake.

Reply:
Moved to Cake/Utilities/FastHash.cs